packer/linux/scripts/install-docker.sh

```diff
@@ -6,28 +6,46 @@ DOCKER_RELEASE="stable"
 DOCKER_COMPOSE_VERSION=1.28.6
 MACHINE=$(uname -m)
 
-# This performs a manual install of Docker.
-
 # Add docker group
 sudo groupadd docker
 sudo usermod -a -G docker ec2-user
 
-# Manual install ala https://docs.docker.com/engine/installation/binaries/
-curl -Lsf -o docker.tgz "https://download.docker.com/linux/static/${DOCKER_RELEASE}/${MACHINE}/docker-${DOCKER_VERSION}.tgz"
-tar -xvzf docker.tgz
-sudo mv docker/* /usr/bin
-rm docker.tgz
+# Add the Docker yum repository, but override the url to get CentOS 7 not
+# CentOS 2 (the default with Amazon Linux 2 applied to the interpolation)
+sudo yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+sudo yum-config-manager --setopt="docker-ce-stable.baseurl=https://download.docker.com/linux/centos/7/$MACHINE/stable" --save
```

Comment:
The default interpolation of this repo file gives us `CentOS 2` because of Amazon Linux 2, this fixes up the URL to work